File: src/Forjix.Application/Features/Management/ManagementModels.cs
namespace Forjix.Application.Features.Management;

public sealed record LookupItem(Guid Id, string Name);
public sealed record UserItem(Guid Id, string Name, string Email, bool IsActive, DateTimeOffset CreatedAt, IReadOnlyList<LookupItem> Groups);
public sealed record SaveUserRequest(string Name, string Email, string? Password, bool IsActive, IReadOnlyList<Guid> RoleIds);
public sealed record RoleItem(Guid Id, string Name, string? Description, bool IsSystem, DateTimeOffset CreatedAt, IReadOnlyList<string> Permissions);
public sealed record SaveRoleRequest(string Name, string? Description, IReadOnlyList<string> Permissions);
public sealed record PermissionItem(string Code, string Name, string Module);
public sealed record AuditItem(long Id, DateTimeOffset OccurredAt, string? UserName, string Action, string Entity, string? EntityId, string? Details);
public sealed record CategoryItem(Guid Id, string Name, string? Description, bool IsActive, DateTimeOffset CreatedAt, DateTimeOffset UpdatedAt);
public sealed record SaveCategoryRequest(string Name, string? Description, bool IsActive);
public sealed record ProductItem(Guid Id, Guid CategoryId, string CategoryName, string Name, string Sku, string? Barcode, decimal SalePrice, decimal CostPrice, decimal MinimumStock, bool IsActive, DateTimeOffset CreatedAt, DateTimeOffset UpdatedAt, string RowVersion);
public sealed record SaveProductRequest(Guid CategoryId, string Name, string Sku, string? Barcode, decimal SalePrice, decimal CostPrice, decimal MinimumStock, bool IsActive, string? RowVersion);

File: web/forjix-web/src/app/core/api/management.models.ts
export interface LookupItem { id: string; name: string; }
export interface UserItem { id: string; name: string; email: string; isActive: boolean; createdAt: string; groups: LookupItem[]; }
export interface SaveUserRequest { name: string; email: string; password?: string | null; isActive: boolean; roleIds: string[]; }
export interface RoleItem { id: string; name: string; description?: string; isSystem: boolean; createdAt: string; permissions: string[]; }
export interface SaveRoleRequest { name: string; description?: string | null; permissions: string[]; }
export interface PermissionItem { code: string; name: string; module: string; }
export interface AuditItem { id: number; occurredAt: string; userName?: string; action: string; entity: string; entityId?: string; details?: string; }
export interface CategoryItem { id: string; name: string; description?: string; isActive: boolean; createdAt: string; updatedAt: string; }
export interface SaveCategoryRequest { name: string; description?: string | null; isActive: boolean; }
export interface ProductItem { id: string; categoryId: string; categoryName: string; name: string; sku: string; barcode?: string; salePrice: number; costPrice: number; minimumStock: number; isActive: boolean; createdAt: string; updatedAt: string; rowVersion: string; }
export interface SaveProductRequest { categoryId: string; name: string; sku: string; barcode?: string | null; salePrice: number; costPrice: number; minimumStock: number; isActive: boolean; rowVersion?: string | null; }
